### PR TITLE
feat: create db search

### PR DESCRIPTION
**Pull request by Romeo1928**

### CODE REVIEW

File: src/api/setDoc.ts [hunk 1]
```diff
@@ -21,17 +21,20 @@ export async function saveUserToDb(
...
       id: user.uid,
       role,
-      avatarUrl,
     };
+    if (phoneNum !== undefined) {
+      userData.phoneNum = phoneNum;
+    }
+    if (avatarUrl !== undefined) {
+      userData.avatarUrl = avatarUrl;
+    }
```
Comment (AdammmVV) [inline src/api/setDoc.ts]:
```suggestion
    if (!phoneNum) {
      userData.phoneNum = phoneNum;
    }
    if (!avatarUrl) {
      userData.avatarUrl = avatarUrl;
    }
```

File: src/components/shared/ComboboxSearch/ComboboxSearch.tsx [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+
+  const onChangeHandler = (e: ChangeEvent<HTMLInputElement>) => {
+    combobox.openDropdown();
+    combobox.updateSelectedOptionIndex();
+    setSearchQuery(e.currentTarget.value.trim());
+  };
+
+  const itemClickHandler = (user: User) => {
+    console.log(user);
+  };
```
Comment (AdammmVV) [inline src/components/shared/ComboboxSearch/ComboboxSearch.tsx]: what does this function do?

File: src/components/shared/ComboboxSearch/ComboboxItems.tsx [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+
+interface IComboboxItems {
+  children: ReactNode;
+}
+
+export const ComboboxItems: FC<IComboboxItems> = ({ children }) => {
+  return <>{children}</>;
+};
+
+export default ComboboxItems;
```
Comment (AdammmVV) [inline src/components/shared/ComboboxSearch/ComboboxItems.tsx]: it's not entirely clear what this is for

File: src/layout/header/HeaderInputSearch/HeaderInputSearch.tsx [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+      <Group>
+        <Avatar radius="xl" size="md" src={user.avatarUrl} />
+
+        <Box>
+          <Text>{user.login}</Text>
+          <Text size="xs">{user.firstName}</Text>
+        </Box>
+      </Group>
+    </Combobox.Option>
+  ));
```
Comment (AdammmVV) [inline src/layout/header/HeaderInputSearch/HeaderInputSearch.tsx]: put the map in the markup and create a component for the item

File: src/layout/header/AuthenticatedContent/AuthenticatedContent.tsx [hunk 5]
```diff
@@ -1,20 +1,25 @@
 import { FC } from 'react';
 import { AppShell, Divider, Group, Paper } from '@mantine/core';
 import { Navbar } from 'layout/navbar/NavBar';
-import { InputSearch } from 'components/shared/search/InputSearch';
+// import { InputSearch } from 'components/shared/search/InputSearch';
```
Comment (AdammmVV) [inline src/layout/header/AuthenticatedContent/AuthenticatedContent.tsx]: remove unused comments here and throughout the application

File: src/api/setDoc.ts (hunk 1 above)
Comment (Romeo1928) [reply src/api/setDoc.ts]: when !phoneNum can't register user in db

**Pull request merged**